iris-mpc-store/src/lib.rs

```diff
@@ -290,21 +290,43 @@ DO UPDATE SET right_code = EXCLUDED.right_code, right_mask = EXCLUDED.right_mask
         Ok(())
     }
 
+    async fn set_sequence_id(
+        &self,
+        id: usize,
+        executor: impl sqlx::Executor<'_, Database = Postgres>,
+    ) -> Result<()> {
+        sqlx::query("SELECT setval(pg_get_serial_sequence('irises', 'id'), $1 + 1, false)")
+            .bind(id as i64)
+            .execute(executor)
+            .await?;
+        Ok(())
+    }
+
     pub async fn rollback(&self, db_len: usize) -> Result<()> {
+        let mut tx = self.pool.begin().await?;
+
         sqlx::query("DELETE FROM irises WHERE id > $1")
             .bind(db_len as i64)
-            .execute(&self.pool)
+            .execute(&mut *tx)
             .await?;
 
-        // We also need to reset the sequence to avoid gaps in the IDs.
-        sqlx::query("SELECT setval(pg_get_serial_sequence('irises', 'id'), $1 + 1, false)")
-            .bind(db_len as i64)
-            .execute(&self.pool)
-            .await?;
+        self.set_sequence_id(db_len, &mut *tx).await?;
 
+        tx.commit().await?;
         Ok(())
     }
 
+    pub async fn set_irises_sequence_id(&self, id: usize) -> Result<()> {
+        self.set_sequence_id(id, &self.pool).await
+    }
+
+    pub async fn get_irises_sequence_id(&self) -> Result<usize> {
+        let id: (i64,) = sqlx::query_as("SELECT currval(pg_get_serial_sequence('irises', 'id'))")
```

Comment:
when I tried this manually in pgAdmin, I got "currval has not been defined in this session" error. Then I used below query to get it working:

`SELECT last_value FROM "SMPC_prod_1".irises_id_seq;`

Let's test in stage and we can use this one if we see any errors 👍

Reply:
hmm then let's just go your proposed way directly?

Reply:
do we need to use the schema though?

Reply:
I don't think schema is needed here as we set the search path to schema already. but i needed it in pgAdmin